### PR TITLE
fix(drawer): drawer update - FE-2798

### LINKED ISSUE #2918: Drawer component: Sidebar is not positioning elements at the top
### Current behaviour

The sidebar in the Drawer component has a margin top applied and causes misalignment with the right hand side.

### Expected behaviour

I'd expect to see the heading in the sidebar inline with the heading in the right hand side.

### Reproducible example

https://codesandbox.io/s/sad-moore-rfqjv?file=/src/index.js

### Suggested solution(s)

Looks like the margin top of 60px is causing this issue.
https://github.com/Sage/carbon/blob/master/src/components/drawer/drawer.style.js#L9

I think there does need to be some consideration that any content that was to reach the chevron would be wrapped?

### Additional context

Initial designs have the chevron hidden

### Your environment

| Software         | Version(s) |
| ---------------- | ---------- |
| carbon-react     | 19.0.0
| carbon-factory   | N/A
| react-scripts    | 3.4.1
| React            | 16.13.1
| Browser          | All
| npm              | 6.14.4
| Operating System | Mac

### PR DESCRIPTION
**Pull request by damianOfKyan**
### Proposed behaviour
Fixes #2918 

Various improvements to the Drawer component:
1. added a new `backgroundColor` prop to control background-color of the sidebar
2. added a new `title` prop to allow user to setup Heading of the sidebar
3. Drawer is now open/expanded by default
4. Various styling updates.

drawer-imp

### Current behaviour
N/A

### Checklist

- [x] Screenshots are included in the PR
<del>- [] Carbon implementation and Design System documentation are congruent
- [x] All themes are supported
- [x] Commits follow our style guide
- [x] Unit tests added or updated
- [x] Cypress automation tests added or updated
- [x] Storybook added or updated
- [x] Typescript `d.ts` file added or updated

### Testing instructions
http://localhost:9001/?path=/docs/design-system-drawer--controlled

### CODE REVIEW

File: cypress/features/regression/drawer.feature [hunk 1]
```diff
@@ -2,42 +2,46 @@ Feature: Drawer component
...
+    Then Drawer "expanded-width-30-drawer" expandedWidth is set to "287.390625px"
+
+  @positive
+  Scenario: Set expandedWidth to 62%
+    When I click on "expanded-width-62-drawer" Drawers arrow 1 time
+    Then Drawer "expanded-width-62-drawer" expandedWidth is set to "593.953125px"
+
+  @positive
+  Scenario: controlled-drawer Drawer backgroundColor is set to yellow
+    Then Drawer "controlled-drawer" backgroundColor is set to "rgb(255, 240, 0)"
```
Comment (DlgSHi) [inline cypress/features/regression/drawer.feature:43]: `When` step is missing

File: cypress/features/regression/drawer.feature [hunk 2]
```diff
@@ -2,42 +2,46 @@ Feature: Drawer component
...
+
+  @positive
+  Scenario: background-color-red Drawer backgroundColor is set to red
+    Then Drawer "background-color-red-drawer" backgroundColor is set to "rgb(255, 0, 0)"
```
Comment (DlgSHi) [inline cypress/features/regression/drawer.feature:47]: `When` step is missing

File: cypress/features/regression/drawer.feature [hunk 3]
```diff
@@ -2,42 +2,46 @@ Feature: Drawer component
...
+  Scenario: Set expandedWidth to 62%
+    When I click on "expanded-width-62-drawer" Drawers arrow 1 time
+    Then Drawer "expanded-width-62-drawer" expandedWidth is set to "593.953125px"
+
+  @positive
+  Scenario: controlled-drawer Drawer backgroundColor is set to yellow
+    Then Drawer "controlled-drawer" backgroundColor is set to "rgb(255, 240, 0)"
+
+  @positive
+  Scenario: background-color-red Drawer backgroundColor is set to red
```
Comment (DlgSHi) [inline cypress/features/regression/drawer.feature:46]:
```suggestion
  Scenario: Drawer backgroundColor is set to red
```

File: cypress/features/regression/drawer.feature [hunk 4]
```diff
@@ -2,42 +2,46 @@ Feature: Drawer component
...
+    When I click on "expanded-width-30-drawer" Drawers arrow 1 time
+    Then Drawer "expanded-width-30-drawer" expandedWidth is set to "287.390625px"
+
+  @positive
+  Scenario: Set expandedWidth to 62%
+    When I click on "expanded-width-62-drawer" Drawers arrow 1 time
+    Then Drawer "expanded-width-62-drawer" expandedWidth is set to "593.953125px"
+
+  @positive
+  Scenario: controlled-drawer Drawer backgroundColor is set to yellow
```
Comment (DlgSHi) [inline cypress/features/regression/drawer.feature:42]:
```suggestion
  Scenario: Drawer backgroundColor is set to yellow
```

File: cypress/features/regression/drawer.feature [hunk 5]
```diff
@@ -2,42 +2,46 @@ Feature: Drawer component
   I want to test Drawer component
```
Comment (DlgSHi) [inline cypress/features/regression/drawer.feature:2]: If we are testing it in `Test` directory this file also needs to be moved to `Test` in `cypress`

File: cypress/features/regression/drawer.feature (hunk 1 above)
Comment (damianOfKyan) [reply cypress/features/regression/drawer.feature]: fixed

File: cypress/features/regression/drawer.feature (hunk 2 above)
Comment (damianOfKyan) [reply cypress/features/regression/drawer.feature]: fixed

File: cypress/features/regression/drawer.feature (hunk 3 above)
Comment (damianOfKyan) [reply cypress/features/regression/drawer.feature]: fixed

File: cypress/features/regression/drawer.feature (hunk 4 above)
Comment (damianOfKyan) [reply cypress/features/regression/drawer.feature]: fixed

File: cypress/features/regression/drawer.feature (hunk 5 above)
Comment (damianOfKyan) [reply cypress/features/regression/drawer.feature]: fixed

**Comment by jamime** [comment]
@damianOfKyan does this fix FE-2790 / #2918 ? If so please can you update the commit, another dev has just started on #2918.

File: src/components/drawer/drawer.component.js [hunk 6]
```diff
@@ -149,12 +179,19 @@ Drawer.propTypes = {
   /* The (% or px) width of the expanded sizebar  */
   expandedWidth: PropTypes.string,
   /** Duration of a animation */
-  animationDuration: PropTypes.string
+  animationDuration: PropTypes.string,
+  /** Sets color of sidebar's background */
+  backgroundColor: PropTypes.string,
+  /** Sets title heading of sidebar's content */
+  title: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
+  /** Enables expand/collapse button that controls drwawer */
```
Comment (DipperTheDan) [inline src/components/drawer/drawer.component.js:187]: Spelling mistake. Should be "drawer".

Comment (damianOfKyan) [reply src/components/drawer/drawer.component.js]: fixed

**Comment by DipperTheDan** [comment]
I think their maybe an issue with your "With Control usage" demo in the MDX file. See the attached GIF.
Drawer Controls Demo Bug

File: cypress/features/regression/designSystem/drawer.feature [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  Background: Open Drawer component page
+    Given I open Design Systems controlled "Drawer" component docs page
+
+  @positive
+  Scenario: Colapsing and Expanding Drawer with Controls
+    When I click on "with-controls" Drawers arrow 1 time
+    Then Drawers "with-controls" sidebar should have class closed
+      And toggle "with-controls" Drawers icon switched orientation to closed
+      And Drawer "with-controls" sidebar text is not visible
+    When I click on "with-controls" Drawers arrow 1 time
```
Comment (DlgSHi) [inline cypress/features/regression/designSystem/drawer.feature:13]: those tests should be separate

Comment (damianOfKyan) [reply cypress/features/regression/designSystem/drawer.feature:13]: I disagree, we need to test the scenario where two clicks are carried out on the toggle button/icon to see the drawer behaviour opening -> open and then closing -> closed.
It is not the same as if I were to render drawer opened and click once to close or opposite.

Comment (damianOfKyan) [reply cypress/features/regression/designSystem/drawer.feature]: fixed.

**Pull request merged**